pandas/tests/plotting/test_frame.py

```diff
@@ -1030,6 +1030,43 @@ def test_plot_scatter(self):
         axes = df.plot(x='x', y='y', kind='scatter', subplots=True)
         self._check_axes_shape(axes, axes_num=1, layout=(1, 1))
 
+    @pytest.mark.slow
+    def test_if_scatterplot_colorbar_affects_xaxis_visibility(self):
+        random_array = np.random.random((1000,3))
+        df = pd.DataFrame(random_array,columns=['A label','B label','C label'])
+
+        ax1 = df.plot.scatter(x='A label', y='B label')
+        ax2 = df.plot.scatter(x='A label', y='B label', c='C label')
+
+        assert all([vis[0].get_visible() == vis[1].get_visible() for vis in
```

Comment:
Just do the `assert`, no error message. pytest will ensure that if the test fails, we still get a meaningful error message.

Reply:
Further, I think you can just do

```python
vis1 = [vis.get_visible() for vis in ax1.xaxis.get_minorticklabels()]
vis2 = [vis.get_visible() for vis in ax2.xaxis.get_minorticklabels()]

assert vis1 == vis2
```

As written, if one happens to be shorter than the other, we wouldn't know because the `zip` would just stop.